File: backend/src/main/java/Presentacion/DTOS/Usuarios/Login/LoginResponse.java
package Presentacion.DTOS.Usuarios.Login;

import Dominio.Entity.Roles.Roles;
import Dominio.Entity.Usuario;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class LoginResponse {
    private String token;
    private String refreshToken;
    private String username;
    private String nombre;
    private String email;
    private String apellido;
    private String rol;
    private boolean verificado;

    public static LoginResponse fromEntity(Usuario usuario){
        if(usuario == null){
            return null;
        }

        return LoginResponse.builder()
                .token(usuario.getToken())
                .refreshToken(usuario.getRefreshToken())
                .username(usuario.getUsername())
                .apellido(usuario.getApellido())
                .nombre(usuario.getNombre())
                .rol(usuario.getRole().name())
                .verificado(usuario.isVerificado())
                .build();
    }
}

File: frontend/lib/models/DTOS/Login/loginResponse.dart
class LoginResponse {
  final String token;
  final String refreshToken;
  final String username;
  final String nombre;
  final String? apellido;
  final String email;
  final String rol;

  LoginResponse({
    required this.token,
    required this.refreshToken,
    required this.username,
    required this.nombre,
    this.apellido,
    required this.email,
    required this.rol,
  });

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      token: json['token'] ?? '',
      refreshToken: json['refreshToken'] ?? '',
      username: json['username'] ?? '',
      nombre: json['nombre'] ?? '',
      apellido: json['apellido'],
      email: json['email'] ?? '',
      rol: json['rol'] ?? '',
    );
  }
}
